sw/applications/example_timer_sdk/main.c

```diff
@@ -25,32 +25,59 @@
     #define PRINTF(...)
 #endif
 
+void __attribute__((aligned(4), interrupt)) handler_irq_timer(void) {
+    timer_arm_stop();
+    timer_irq_clear();
+    return;   
+}
+
 int main(){
-    uint8_t i = 0;
-    uint32_t cpu_cycles;
+    uint32_t i = 0;
+    uint32_t timer_cycles;
 
-    timer_init();               // Init the timer SDK
+    timer_init();               // Init the timer SDK for clock cycles
     timer_start();              // Start counting the time
-    cpu_cycles = timer_stop();  // Stop counting the time
-    PRINTF("0 NOPs:\t%d cc\n\r", cpu_cycles );
+    timer_cycles = timer_stop();  // Stop counting the time
+    PRINTF("0 NOPs:\t%d cc\n\r", timer_cycles );
     
-    timer_start();              
+    timer_start();     
     asm volatile ("nop");
-    cpu_cycles = timer_stop();  
-    PRINTF("1 NOP:\t%d cc\n\r", cpu_cycles );
+    timer_cycles = timer_stop();  
+    PRINTF("1 NOP:\t%d cc\n\r", timer_cycles );
     
     timer_start();              
     asm volatile ("nop");
     asm volatile ("nop");
-    cpu_cycles = timer_stop();  
-    PRINTF("2 NOPs:\t%d cc\n\r", cpu_cycles );
+    timer_cycles = timer_stop();  
+    PRINTF("2 NOPs:\t%d cc\n\r", timer_cycles );
 
     timer_start();              
     asm volatile ("nop");
     asm volatile ("nop");
     asm volatile ("nop");
-    cpu_cycles = timer_stop();  
-    PRINTF("3 NOPs:\t%d cc\n\r", cpu_cycles );
-    
+    timer_cycles = timer_stop();  
+    PRINTF("3 NOPs:\t%d cc\n\r", timer_cycles );
+
+    enable_timer_interrupt();   // Enable the timer machine-level interrupt
+
+    timer_init();
+    timer_irq_enable();
+    timer_arm_start(1000);
+    timer_start();              
+    asm volatile ("wfi");       // Wait for interrupt
+    timer_cycles = timer_stop();  
+    PRINTF("wait ARM timer 1000 (%d)\n\r", timer_cycles );
+
+    timer_init_us();           // Init the timer SDK for microseconds
+    timer_start(); 
+    for(i = 0; i < 1000; i++){
+        asm volatile ("nop");
+    }
+    PRINTF("Microseconds for 1000 NOPs:\t%d μs\n\r", timer_stop() );
+
+    PRINTF("Wait 5 seconds\n\r");
+    timer_wait_ms(5000);       // Wait for 5 seconds
+    PRINTF("Done\n\r");
+
```

Comment:
there must be a failing or passing condition, return EXIT_SUCCESS whatever happens does not help